### ISSUE TITLE
[3.3.5/Core/PetAI/] Pets will not be summoned back if the teleportation is to the same map

### ISSUE THREAD

**Issue by soniczhang11** [TrinityCore/TrinityCore#23325]
**Description:**

Pets always get stuck in corners and after Hearthstone warlock's pets will be dismissed.  

**Current behaviour:**

It seem like the mmaps or vmaps LOS block pets' sights or waypoints.
found a solution to this issue 7 years ago: ac442aa6195b6e0a39127f5df540563f0b77eacb, but cannot sure whether it still works.

**Expected behaviour:**

1)Maybe remove the LOS of pets, or give them a better AI? 
2)keep warlocks' pets after Hearthstone. 

**Steps to reproduce the problem:**

1. enabel the vmapsLOS and mmaps in worldserver.conf. 
2. Use GM account to create a warlock and levelup.
3. Summon a pet and go in and go out some structurally complex buildings and observe the pets.
4. Use Hearthstone and then pet will be dismissed

**Branch(es):**

3.3.5

**TC rev. hash/commit:** 

16bf11ab20fcd5e3984dd90d011d13f56288d1ea

**Operating system:** CentOs

**Comment by illfated** [comment]
Getting the pet stuck can be replicated using hunter pets too. Replication steps:
1. Go to Booty Bay (by boat, ride into the area or use `.tele booty`)
2. Approach the Salty Sailor Tavern from the docks level
3. Dismount in good time before the Tavern entrance.
4. Stop shortly after entering, to make the pet stop at the threshold.
5. Continue walking through the building.

You may need to go back out and in again to enable your pet to enter the building.
This can also happen near any stair or threshold in instances like Shadowfang Keep.
Depending on fps or LOS, the pet may be dismissed on HS.

Last tested on TrinityCore rev. bfffa5d45e9d 2019-05-27 03:21:36 +0200 (3.3.5 branch) (Unix, RelWithDebInfo, Static)
(tcubuntu.northeurope.cloudapp.azure.com)

**Comment by soniczhang11** [comment]
I found if I set mmap.enablePathFinding = 0 in worldserver.conf, this issue will disappear, but then mobs will ignore landform and move by a shortcut.

**Comment by illfated** [comment]
Yes, that is pretty much the expected outcome of disabling `mmap.enablePathFinding` in worldserver.conf . That is also why it is recommended to enable that setting (and has been for along time).

**Comment by soniczhang11** [comment]
> Yes, that is pretty much the expected outcome of disabling `mmap.enablePathFinding` in worldserver.conf . That is also why it is recommended to enable that setting (and has been for along time).

now the problem is if I want the mobs respect the landform and obstacles I must enable mmap pathfinding. But then pets will get stucked very easily.

**Comment by illfated** [comment]
Maybe the source needs to be updated to make pets follow the owner's path? Then again, it is difficult enough to make pets choose the correct path, even when moving towards or back from their target. Unfortunately, it is far from a trivial matter to fix the pet movement behaviour.

**Comment by jackpoz** [comment]
is it possible to get a video/screenshots of the issue ?

**Comment by Killyana** [comment]
The only issue I can reproduce is:
If you have a pet and use your Hearthstone, if the teleportation happen to the same current map, the pet will not be summoned back. It also happen with portal teleportation like:
From Wintergrasp to Dalaran the pet will not be summoned back (.go game 67278)
From Dalaran to Orgrimmar it will be summoned (.go game 56568 )

This issue doesn't happen with the command .tele

**Comment by illfated** [comment]
> in this case it will be abandoned

Just a minor (but important enough to distinguish) technical detail: "in this case, it will be regarded as **dismissed**."

**Comment by jackpoz** [comment]
About the teleport part, it's probably some different behavior depending on spell effect (maybe "Teleport Unit" with range keep the pet, without range resummon it)

**Comment by CraftedRO** [comment]
While I might be wrong but in my recent tests looks like the teleport on the same map issue have been fixed recently.

**Comment by CraftedRO** [comment]
ah nvm my bad

**Comment by offl** [comment]
Can we close this now?
